lock-api-objects/src/main/java/com/palantir/lock/watch/NoOpLockWatchEventCache.java

```diff
@@ -25,18 +25,17 @@
 @SuppressWarnings("FinalClass") // mocks
 public class NoOpLockWatchEventCache implements LockWatchEventCache {
     public static final LockWatchEventCache INSTANCE = new NoOpLockWatchEventCache();
-    private static final IdentifiedVersion FAKE = ImmutableIdentifiedVersion
-            .of(UUID.randomUUID(), Optional.empty());
+    private static final IdentifiedVersion FAKE = ImmutableIdentifiedVersion.of(UUID.randomUUID(), 0L);
     private static final TransactionsLockWatchEvents NONE = TransactionsLockWatchEvents.failure(
-            LockWatchStateUpdate.snapshot(UUID.randomUUID(), 0L, ImmutableSet.of(), ImmutableSet.of()));
+            LockWatchStateUpdate.snapshot(UUID.randomUUID(), -1L, ImmutableSet.of(), ImmutableSet.of()));
 
     private NoOpLockWatchEventCache() {
         // singleton
     }
 
     @Override
-    public IdentifiedVersion lastKnownVersion() {
-        return FAKE;
+    public Optional<IdentifiedVersion> lastKnownVersion() {
+        return Optional.empty();
```

Comment:
Thinking about this a bit more: I think this is a change in behavior, in that every startTransactions call now will cause timelock to compute a snapshot.

Reply:
I guess snapshotting right now is cheap since there aren't actually going to be events, but this could lead to unintended consequences later on. It seems safer to return `Optional.of(FAKE)`? - you still take a snapshot when there's a leader election, but that's relatively uncommon

Reply:
Exactly, I think snapshotting is cheap so I shouldn't worry about it.

Reply:
Returning FAKE is same as Optional.empty, see impl. I think this is semantically clearer, since we're simply not tracking the version.